x-pack/plugins/lens/public/xy_visualization/to_expression.ts

```diff
@@ -118,8 +119,22 @@ export const buildExpression = (
             },
           ],
           fittingFunction: [state.fittingFunction || 'None'],
-          showXAxisTitle: [state.showXAxisTitle ?? true],
-          showYAxisTitle: [state.showYAxisTitle ?? true],
+          axisTitlesVisibilitySettings: [
+            {
+              type: 'expression',
+              chain: [
+                {
+                  type: 'function',
+                  function: 'lens_xy_axisTitlesVisibilityConfig',
+                  arguments: {
+                    x: [state?.axisTitlesVisibilitySettings?.x ?? true],
+                    yLeft: [state?.axisTitlesVisibilitySettings?.yLeft ?? true],
+                    yRight: [state?.axisTitlesVisibilitySettings?.yRight ?? true],
+                  },
+                },
+              ],
+            },
+          ],
```

Comment:
Because you have changed the expected shape of the `state` object, you need to write a migration for this, or it will fail when upgrading from older versions.

Reply:
You are right @wylieconlon 🙂 We can avoid the migration by just adding a `yRightTitle` and leave the `yTitle` as it is.

Reply:
Normally we would need to migrate, but as this feature was not released yet (all within 7.10), it should be fine.

Reply:
Yes except from the yTitle, but I have reverted it so I think we are ok now

Reply:
Right, my bad.